### ISSUE TITLE
[HOLD] Keep navigation within the RHP when navigating from a report to an expense on the Search page

### ISSUE THREAD

**Issue by shawnborton** [Expensify/App#52939]
**Background:** Users often utilize the Search page to find and access reports through the Right-Hand Pane (RHP). The RHP allows them to view reports without leaving the Search page, providing a seamless and efficient experience.

**Problem:** When users attempt to view an expense from a report opened in the RHP, they are redirected to the Inbox, disrupting their workflow and preventing them from maintaining their original context.

**Solution:** Adjust the RHP navigation stack in Search so that any content accessed from within the RHP remains within it. This change will enable users to navigate from a report to its associated expenses entirely within the RHP, allowing for smooth, uninterrupted exploration of report details without leaving the Search page.

https://github.com/user-attachments/assets/59dcdbb9-1835-47d4-aefd-82b2e43a6d43

**Comment by Kicu** [comment]
This is being actively worked on

**Comment by shawnborton** [comment]
Not overdue, SWM is working on it.

**Comment by shawnborton** [comment]
Good question, thoughts @luacmartins ?

**Comment by luacmartins** [comment]
They are the same. I'll close the other issue in favor of this one, since it has a more structured P/S

**Comment by trjExpensify** [comment]
Coupla' things: 

- We are going to keep this moving forward, yes?
- I'm moving this project into #migrate to track along with this next wave of changes.

**Comment by luacmartins** [comment]
@trjExpensify @shawnborton can you share what the new navigation flows should be in the Search page? That'd give us a better idea if this is still needed or not

**Comment by JmillsExpensify** [comment]
What do you mean by new navigation flows?

**Comment by trjExpensify** [comment]
The redesign, report in the main pane etc.

**Comment by luacmartins** [comment]
Basically what navigating from an expense/report row to a report in the main pane to an expense in the RHP then clicking any link in the RHP, etc

**Comment by mountiny** [comment]
Native stack PR merged and getting through the follow up bugs, this will unblock the bottom tab refactor and these changes too

**Comment by luacmartins** [comment]
>That being said, I do think it makes sense to keep the user in the RHP if they are navigating into a thread inside of an expense or something like that.

So we'd keep the user in the RHP if they clicked any link in the RHP chat?

**Comment by shawnborton** [comment]
I wouldn't say "any link" but we'd want to keep them in the RHP if they were drilling down into a thread or something.

It gets a bit tricky though because in theory, you could first open up an expense in the RHP from the Search > All page. But then from the expense, you could go up a level and view the report. Maybe it's okay to keep that report in the RHP, but then we have a consistency where sometimes the report can be opened in the main content pane and sometimes it can be opened up in the RHP. 

I think the best path forward might just be to try to keep the user in the RHP if they are navigating around within the RHP in relation to workspace chat/report/expense views.

**Comment by shawnborton** [comment]
Going to throw a HOLD on this one while we sort out some of the high-level discussions around Search V3. Sorry for the change of plans, but let's make sure we attack this one with alignment.

**Comment by shawnborton** [comment]
Still holding

**Comment by shawnborton** [comment]
Still holding

**Comment by shawnborton** [comment]
Still holding

**Comment by shawnborton** [comment]
Still holding...

**Comment by shawnborton** [comment]
Still holding.

**Comment by luacmartins** [comment]
Still on hold

**Comment by shawnborton** [comment]
I wonder if we should just close this out in favor of my expense previews project? Feels like there is a lot of overlap here at least.

**Comment by luacmartins** [comment]
Yea, we can close this I think

**Comment by shawnborton** [comment]
Cool, closing!

**Comment by mountiny** [comment]
Reopening as we will want to do this after the navigation changes are in place cc @WojtekBoman 

We will need to check if there are any considerations that changes

**Comment by mountiny** [comment]
Wojtek is looking into this
